crates/net/network/src/transactions/fetcher.rs

```diff
@@ -466,6 +458,10 @@ impl TransactionFetcher {
 
             // vacant entry
 
+            if is_tx_bad_import(hash) {
+                return false
+            }
```

Comment:
we're doing this here because this check might not be hit if the hash already exists but is not bad, right?

this makes sense and should be more efficient than filtering before calling this function most of the time

Reply:
upon inserting an entry into bad imports, it's also removed from the tx fetcher

https://github.com/paradigmxyz/reth/blob/c696b00fd02012e5fd42c0066b1e58c88a9f9bf5/crates/net/network/src/transactions/mod.rs#L1032

but, since the bad imports cache is limited and (for now) the cache for all hashes in tx fetcher `hashes_fetch_inflight_and_pending_fetch` is unlimited...though unlikely, it can happen that a bad import has been evicted from bad imports by the time it is announced. so, let's move that check to run regardless if the entry in `hashes_fetch_inflight_and_pending_fetch` is vacant or occupied.